### DISCUSSION TITLE
Suggested modification to current roadmap

### DISCUSSION THREAD

**Discussion by ilan-schemoul** [nvim-treesitter/nvim-treesitter#6718] [General]
Hello,

I have to open a new issue as the main one is locked #4767

I would like to offer a suggestion/question : do we really need WASM support as a requirement for 1.0 ? 

1) If you are talking about this issue https://github.com/neovim/neovim/issues/23579 it doesn't say anywhere that it will be released for 0.11. Maybe you're talking about something else I am unaware then I'd be interested to read about it.
2) Multiple treesitter plugins are in a state of flux where they wait for the major refactoring that is the removal of the plugin system and halt/slow progress waiting for this uncertain major refactoring they will have to undergo/have to maintain two branches.

Questions : 
1) Will WASM support be a breaking change ? i.e: if we have a 1.0 does wasm support mean 2.0 ?
2) Can you provide the community information with why WASM support became a requirement ? (I am curious because personally compilation of TS takes like a second so I don't care in the slightest but I guess there are concerns about users not having gcc etc ? Or maybe something else ?). I understand why WASM can be helpful I specifically want to know why making it a 1.0 requirement.

My understanding is that nvim was partly created to offer an alternative to vim's rigid "benevolant dictator" approach with a more inclusive gouvernance therefore I feel like not locking issues and sharing more about critical issue would be great and more in line with the original neovim's philosophy.

**Comment by clason** [comment]
We are talking specifically about wasm parsers for which there is already a Neovim PR up. This will be the main parser distribution according to upstream policy, and this will be the default for stable parsers in this plugin as well (otherwise tree-sitter will be a hard requirement, and we don't want that; the old "try every compiler and compiler options" is just not sustainable). 

(And this is not Neovim, just a random plugin. You can write your own plugin if you want things done differently.)

**Reply by ilan-schemoul** [reply]
> We are talking specifically about wasm parsers for which there is already a Neovim PR up. This will be the main parser distribution according to upstream policy, and this will be the default for stable parsers in this plugin as well (otherwise tree-sitter will be a hard requirement, and we don't want that; the old "try every compiler and compiler options" is just not sustainable).

Thank you, that answer some of my questions.

But still my main question is can you consider shipping 1.0 to do the major architecture major refactoring and then subsequently do the WASM update ? Or is that a no go ?

Because right now progress is halted on great plugins such as TS-textobject waiting for this package to be 1.0 and this package waits for the WASM parser PR to be done and then will wait for the release to include that PR (and release date are hard to predict). It is a long dependency chain right now and I'm wondering if things could be simplified a bit ?

> (And this is not Neovim, just a random plugin. You can write your own plugin if you want things done differently.)

This is anything but a random package. It is an important package and other major packages (such as TS-textobject which is important in my workflow for example) waits for the 1.0 of this package.

I mention transparency because I feel like when we talk about tools used by tens of thousands of users such as nvim's TS it would be positive for us to understand a bit more what's going under the hood. You are influential in many plugins and in core and we would gain a lot if you were to decide to be a bit more transparent when you take decisions impacting our tools. I thank you for your work this is just my input which of course you are free to ignore.

**Reply by clason** [reply]
Yes, wasm parsers will be a breaking update, and I only want to make one such major break. And the whole point of this is that nvim-treesitter should _not_ be a major dependance for other plugins (that I do not maintain). I am as transparent about this as I can be (hence the roadmap); please accept that all this is best-effort, and plans will have to change based on changing situations. Currently, the roadmap is blocked by tree-sitter changes, which I hope to be resolved in the next version or two over the summer.

And just to be clear, both nvim-treesitter and nvim-treesitter textobjects are working just fine; I understand being frustrated that your PR doesn't get merged, but it's not like the current situation is an actual problem.

**Reply by ilan-schemoul** [reply]
Alright that's a very fair take. Thanks a lot for all this input.
My point is beyond my small PR which doesn't matter too much but more about TS's importance in my workflow (and many others) so I was wondering how the ecosystem can move forward. TS texobjects works but like the rest of the ecosystem would gain from TS and nvim-TS to become stable. There is multiple things that would need improvement but I also wait for stability and 1.0.
If WASM is important and would be a breaking change then it makes sense to wait for it of course.

> Currently, the roadmap is blocked by tree-sitter changes, which I hope to be resolved in the next version or two over the summer.

Nvim 0.11 current expected release date is end of October. Release date rarely are precise at an early stage so yeah we'd have to probably wait next year for everything to move forward. Which was my reason to ask more questions about the roadmap. 

Thanks a lot for all these answers again.
